File: app/src/main/java/pl/kopsoft/pczplan/models/SchoolWeekSchedule.java
package pl.kopsoft.pczplan.models;


import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class SchoolWeekSchedule implements Serializable {
    public List<SchoolDaySchedule> DaySchedules;

    public SchoolWeekSchedule() {
        DaySchedules = new ArrayList<>();
    }

    public void TrimAll() {
        for (int i = 0; i < DaySchedules.size(); i++) {
            if (DaySchedules.get(i) != null) {
                DaySchedules.get(i).TrimEmptySubjects();
            }
        }
    }
}

File: app/src/main/java/pl/kopsoft/pczplan/models/SchoolWeekSchedule.kt
package pl.kopsoft.pczplan.models


import java.io.Serializable

class SchoolWeekSchedule : Serializable {
    val daySchedules: ArrayList<SchoolDaySchedule> = arrayListOf()

    fun trimAll() {
        for (i in daySchedules.indices) {
            daySchedules[i].trimEmptySubjects()
        }
    }

    fun addDay(day: SchoolDaySchedule) {
        daySchedules.add(day)
    }
}
